ls1intum/Artemis
# [Feature] Use JPlag for text exercises plagiarism detection

### Checklist

### Motivation and Context
Plagiarism detection for text exercises does not support JPlag yet. However, our current plagiarism detection algorithm is quite slow so JPlag might be the preferable option. The user should be able to decide which strategy/algorithm to use when triggering the plagiarism detection.

### Description
The Artemis client application can now call the `/check-plagiarism` endpoint for text exercises and pass an optional `strategy` parameter to specify the strategy to use for text exercise comparison. If no `strategy` parameter is specified, the algorithm will default to the current implementation.

Example request to trigger the plagiarism detection using JPlag:
`GET http://localhost:9000/api/text-exercises/13/check-plagiarism?strategy=JPlag`

**Note**: As discussed with @krusche, this PR does not contain tests for the new functionality. These tests will be implemented in a follow-up PR.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a text exercise
4. Create similar submissions for the text exercise on multiple student accounts
5. Log in as the course instructor and navigate to the exercise details view

Make sure that:
1. Clicking on "Check Plagiarism" now offers a third option called "JPlag"
2. Clicking on "JPlag" correctly sets the `strategy=JPlag` parameter in the `GET` request
3. The server responds with a ZIP file containing the JPlag output (a bunch of HTML files, no need to check those)

### Screenshots
<img width="644" alt="Screenshot 2020-10-05 at 12 14 38" src="https://user-images.githubusercontent.com/21973908/95070663-bbfd3b00-0708-11eb-968a-0f6f7ecf32ca.png">
